packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,365 @@
+      setSelectedFiles([...selectedWithoutReplacedFiles, ...replacedFiles, ...newFiles]);
+      setSuccessText(getAddSuccessMessage(language));
+    } else {
+      setSuccessText(undefined);
+    }
+  };
+
+  const onFilesChange = (files: File[]) => {
+    if (multiple) {
+      handleMultipleChange(files);
+    } else {
+      handleSingleFileChange(files);
+    }
+  };
+
+  const onRemoveFileFromList = (fileToRemove: File, index: number) => {
+    const selectedFilesWithoutRemoved = selectedFiles.filter(
+      (file: File) => !isEqualFileBy(['name', 'type', 'size', 'lastModified'], file, fileToRemove),
+    );
+    setSelectedFiles(selectedFilesWithoutRemoved);
+    setSuccessText(getRemoveSuccessMessage(language));
+
+    if (selectedFilesWithoutRemoved.length > 0) {
+      fileListFocusIndexRef.current = index > 0 ? index - 1 : 0;
+    } else if (fileListRef.current) {
+      fileListRef.current.focus();
+    }
+  };
+
+  const onDragEnter = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(true);
+  };
+
+  const onDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(false);
+  };
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedFiles);
+    }
+    // Clear input value on every change to ensure it triggers a onChange event when files are added
+    resetFileInputValue();
+  }, [selectedFiles, onChange]);
+
+  // Compose aria-describedby attribute
+  const ariaDescribedBy: string = [
+    comboseAriaDescribedBy(id, helperText, errorText, successText),
+    hasFilesSelected && fileListId,
+  ]
+    .filter((text) => !!text)
+    .join(' ');
+
+  return (
+    <>
+      <InputWrapper {...wrapperProps}>
+        <div className={styles.fileInputContainer}>
+          {hasDragAndDrop && (
+            <>
+              <div
+                aria-hidden
+                onDragEnter={onDragEnter}
+                onDragOver={onDragEnter}
+                onDragLeave={onDragLeave}
+                onDrop={(event: React.DragEvent<HTMLDivElement>) => {
+                  const { dataTransfer }: { dataTransfer: DataTransfer } = event;
+                  onDragLeave(event);
+                  onFilesChange(Array.from(dataTransfer.files));
+                }}
+                className={classNames(styles.dragAndDrop, isDragOverDrop && styles.dragAndDropActive)}
+                ref={dropAreaRef}
+              >
+                <div className={styles.dragAndDropLabel}>
+                  <IconUpload aria-hidden />
+                  <span className={styles.dragAndDropLabelText}>{dragAndDrop.label}</span>
+                </div>
+              </div>
+              <div className={styles.dragAndDropHelperText}>{dragAndDrop.helperText}</div>
+            </>
+          )}
+          <div className={styles.fileInputWrapper}>
+            <Button
+              aria-hidden
```

Comment:
Is this aria-hidden necessary?

Reply:
Yes, two inputs side-by-side complicate things with assistive technology.